io/disk/ltp_fs.py

```diff
@@ -0,0 +1,116 @@
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2016 IBM
+# Author: Harsha Thyagaraja <harshkid@linux.vnet.ibm.com>
+#
+# Based on code by Martin Bligh <mbligh@google.com>
+# copyright 2006 Google, Inc.
+# https://github.com/autotest/autotest-client-tests/tree/master/ltp
+
+
+import os
+from avocado import Test
+from avocado import main
+from avocado.utils import build
+from avocado.utils import process, archive
+from avocado.utils.software_manager import SoftwareManager
+
+
+class ltp(Test):
+    '''
+    LTP (Linux Test Project) testsuite
+    '''
+    def setUp(self):
+        '''
+        To check and install dependencies for the test
+        '''
+        sm = SoftwareManager()
+        deps = ['gcc', 'make', 'automake', 'autoconf']
+        for package in deps:
+            if not sm.check_installed(package) and not sm.install(package):
+                self.error(package + ' is needed for the test to be run')
+        self.disk = self.params.get('disk')
+        self.mount_point = self.params.get('mount_point', default=self.srcdir)
+        self.script = self.params.get('script')
+        self.fs = self.params.get('fs', default='ext4')
+        self.args = self.params.get('args', default='').split(',')
+        self.log.info("creating ext4 filesystem on disk %s" % self.disk)
+        create_fs = "mkfs.%s %s" % (self.fs, self.disk)
+        if process.system(create_fs, shell=True, ignore_status=True) != 0:
+            self.fail("Creating filesystem %s on\
+                       %s failed" % (self.fs, self.disk))
+        self.log.info("Mounting disk %s to mount location %s"
+                      % (self.disk, self.mount_point))
+        mounting = "mount %s %s" % (self.disk, self.mount_point)
+        if process.system(mounting, shell=True, ignore_status=True) != 0:
+            self.fail("Unable to mount to location  %s" % self.mount_point)
+        url = "https://github.com/linux-test-project/ltp/\
+               archive/master.zip"
+        tarball = self.fetch_asset("ltp-master.zip",
+                                   locations=[url], expire='7d')
+        archive.extract(tarball, self.srcdir)
+        ltp_dir = os.path.join(self.srcdir, "ltp-master")
+        os.chdir(ltp_dir)
+        build.make(ltp_dir, extra_args='autotools')
+        ltpbin_dir = os.path.join(ltp_dir, 'bin')
+        os.mkdir(ltpbin_dir)
+        process.system('./configure --prefix=%s' % ltpbin_dir)
+        build.make(ltp_dir)
+        build.make(ltp_dir, extra_args='install')
+
+    def test(self):
+        for val in self.args:
```

Comment:
@narasimhan-v  
Let's consider the following Instance example if we multiplex the args:
args: !mux
    test:
        args: '-s fs_di'
    test: 
        args: '-s fs_inod'
fs: !mux
    ext4:
        fs: 'ext4'
    ext3:
        fs: 'ext3'

So now my script will run for 4 variants: 
variant1: fs_di ext4
variant2: fs_di ext3
variant3: fs_inod ext4
variant4: fs_inod ext3

This causes the execution time to be more.


If we consider my current config in yaml:
args: '-s fs_di,-s fs_inod'
fs: !mux
    ext4:
        fs: ext4
    ext3:
        fs: ext3


We will have 2 variants here instead of 2:
variant1: fs_di, fs_inod ext4
variant2: fs_di, fs_inod ext3

I want the execution time to be cut down with my consideration of 'args' usage